include/drivers/hwinfo.h

```diff
@@ -30,6 +30,18 @@
 extern "C" {
 #endif
 
+enum reset_cause {
+	RESET_PIN				= BIT(0),
+	RESET_SOFTWARE				= BIT(1),
+	RESET_BROWNOUT				= BIT(2),
+	RESET_POWER				= BIT(3),
+	RESET_WATCHDOG				= BIT(4),
+	RESET_DEBUG				= BIT(5),
+	RESET_SECURITY				= BIT(6),
+	RESET_LOW_POWER_WAKE			= BIT(7),
+	RESET_CPU_LOCKUP			= BIT(8)
+};
```

Comment:
This needs far more docs on what these are suppose to mean.

Reply:
Probably adding a RESET_SOC (ie SoC specific reason).

Reply:
Agreed. I am trying to gather all feedback before doing more work. There was/is some uncertainty about the overall approach.

Reply:
Would RESET_OTHER work better than RESET_SOC for the causes that are not supported by this list?